R/check_suggested.R

```diff
@@ -39,22 +36,36 @@ check_suggested <- function(packages,
     return(FALSE)
   }
 
-  # If TRUE, fail if package is not installed
-  for (pkg in packages) {
-    if (!is_installed(pkg)) {
-      if (is_testing()) {
-        if (is.null(message)) {
-          message <- glue("`{top_level_fun}()` needs the '{pkg}' package.")
-        }
+  # Return early if all packages are installed.
+  if (rlang::is_installed(packages, version = version)) {
+    return(TRUE)
+  }
+
+  # Skip if some packages are not installed when testing
+  # And say which package was not installed.
+  if (is_testing()) {
+    for (pkg in packages) {
```

Comment:
Once we have a logical `installed` vector, we can do `packages[!installed]` and bundle the result in a single `message()` or `skip()` call, removing the `for` loops.

Reply:
A prototype function would be the following.
```r
  installed <- purrr::map2_lgl(packages, version, function(pkg, ver) {
    is_installed(pkg, version = ver)
  }
```

Reply:
Let's get rid of the `version` argument in our function if we don't need it.

Reply:
Yes. and use `check_installed()` syntax instead. I will do this.